### DISCUSSION TITLE
How to create a new module in cosmos-sdk and add it to existing blockchain. Please provide step by step guide.

### DISCUSSION THREAD

**Discussion by coderasha** [cosmos/community#73] [Chain Builders (Cosmos SDK)]
I have been exploring cosmos docs but couldn't find a proper documentation on all the steps to create a new module and add it to the Cosmos-SDK chain with our own customization.

**Comment by julienrbrt** [comment]
Hi, you can use ignite to scaffold the basis of chain and modules or use a module template (https://github.com/cosmosregistry/example) which contains all the necessary bits to have a module working.

Additionally, you can take a look here for step by step instructions: https://tutorials.cosmos.network/hands-on-exercise/0-native/2-build-module.html

**Comment by coderasha** [comment]
is there any youtube tutorial expaining each and everysteo=p in detail?

**Reply by facundomedica** [reply]
I don't know if it's going to be enough detail but you can complement the resources that @julienrbrt posted with this video: https://www.youtube.com/watch?v=9kK9uzwEeOE
